### DISCUSSION TITLE
How do I map an array of ObjectId from one model to another?

### DISCUSSION THREAD

**Discussion by mapringg** [prisma/prisma#17083] [Q&A] [answered]
I have a favorites model that hold an array of object id that map to a report model id. Is it possible to make a relation out of this?

```prisma
model favorites {
  id         String   @id @default(auto()) @map("_id") @db.ObjectId
  account_id String
  favorites  String[] @db.ObjectId
}
```

```prisma
model reports {
  id            String   @id @default(auto()) @map("_id") @db.ObjectId
  account_id    String
  auth_required Boolean
  createdAt     DateTime @db.Date
  from_app      String
  link          String
  name          String
  tags          String[] @db.ObjectId
  updatedAt     DateTime @db.Date
  zr_link       String
}
```

**Comment by nurul3101** [comment, marked as answer]
Hello @mapringg 👋 
Could you try this schema?
In this schema, favorites can have multiple reports.
Also, please note that as favorites is a relation field, it will only be available at Prisma level and not at the database level.

```prisma
model favorites {
  id         String    @id @default(auto()) @map("_id") @db.ObjectId
  account_id String
  favorites  reports[]
}

model reports {
  id            String     @id @default(auto()) @map("_id") @db.ObjectId
  account_id    String
  auth_required Boolean
  createdAt     DateTime   @db.Date
  from_app      String
  link          String
  name          String
  tags          String[]   @db.ObjectId
  updatedAt     DateTime   @db.Date
  zr_link       String
  favorites     favorites? @relation(fields: [favoritesId], references: [id])
  favoritesId   String?    @db.ObjectId
}
```

**Reply by mapringg** [reply]
Thank you for replying.

The field `favorites` tells me which report is my favorite. Currently, I have to query both and then filter them. I was thinking if it was possible to get something like an isFavorite field in the `reports` model, but I think it might not be possible.

**Reply by mapringg** [reply]
I played around with it more and think I kinda get it now, but I'm still not sure if this is the correct way. Currently, I can map the `favoritesId` to the `id` inside the `favorites` model. That result in the `favorites` field updating and I could use the `favoritesId` field that is not null to verify whether my report is favorite or not. Are there any performance improvement by doing this way instead of double query and filter?

**Reply by nurul3101** [reply]
Ideally, if you can achieve the same outcome in a single query, then it should be more performant compared to getting the same outcome by doing two queries. I would recommend you log the MongoDB queries executed by configuring [logging](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) and then understand the query execution plan for both cases through [explain](https://www.mongodb.com/docs/manual/reference/command/explain/#mongodb-dbcommand-dbcmd.explain). That should provide you thorough insights on time taken for each case and also you can potentially improve performance by adding indexes on the fields which are used for filtering.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
